app/code/Magento/Variable/Model/Variable.php

```diff
@@ -162,4 +179,39 @@ public function getVariablesOptionArray($withGroup = false)
         }
         return $variables;
     }
+
+    /**
+     * Check if the given store (or the current one) has the given
+     * variable set with the given text value
+     *
+     * @param string $variableCode
+     * @param string $expectedValue
+     * @param string $type
+     * @param int $storeId
+     *
+     * @return boolean
+     */
+    public function customVariableHasValue(
+        $variableCode,
+        $expectedValue,
+        $type = self::TYPE_TEXT,
+        $storeId = null
+    ) {
+        $status = false;
+
+        if ($storeId === null) {
+            $storeId = $this->storeManager->getStore()->getId();
+        }
+
+        $this->setStoreId($storeId)
+            ->loadByCode($variableCode);
```

Comment:
I totally agree, and actually, the first thing I was looking for was a repository... Didn't find it so I went to the Model with the old Active-Record pattern.
Creating a repository for the _Variable_ entity could be a future task...